tests/platform_tests/test_link_down.py

```diff
@@ -0,0 +1,205 @@
+            else:
+                # Check every interfaces are down on this host every 5 sec until device boots up
+                pytest_assert(wait_until(MAX_TIME_TO_REBOOT, 5, 0, is_link_down, fanout, port),
+                              "Interface {} on {} is still up after {}sec".format(port, hostname, MAX_TIME_TO_REBOOT))
+                             
+        logger.info("All interfaces on {} are {}".format(fanout.hostname, "up" if up else "down"))
+    return True
+
+
+def link_status_on_all_LC(duthosts, localhost, fanouts_and_ports, up=True):
+    """
+    Return:
+        True: all links on all LCs are down
+    """
+    for LC in duthosts.frontend_nodes:
+        link_status_on_host(LC, localhost, fanouts_and_ports, up)
+    logger.info("All interfaces on all linecards are down!")
+    return True    
+
+
+def check_interfaces_and_services_all_LCs(duthosts, conn_graph_facts, xcvr_skip_list):
+    for LC in duthosts.frontend_nodes:
+        check_interfaces_and_services(LC, conn_graph_facts["device_conn"][LC.hostname], xcvr_skip_list)
+
+
+def test_link_down_on_sup_reboot(duthosts, localhost, enum_supervisor_dut_hostname, 
+                                 conn_graph_facts, duts_running_config_facts, 
+                                 fanouthosts, tbinfo, xcvr_skip_list):
+    if len(duthosts.nodes) == 1:
+        pytest.skip("Skip single-host dut for this test")
+
+    duthost = duthosts[enum_supervisor_dut_hostname]
+    hostname = duthost.hostname
+    # Before test, check all interfaces and services are up on all linecards
+    check_interfaces_and_services_all_LCs(duthosts, conn_graph_facts, xcvr_skip_list)
+
+    duts_and_ports = multi_duts_and_ports(duthosts)
+    fanouts_and_ports = fanout_hosts_and_ports(fanouthosts, duts_and_ports)
+
+    # Also make sure fanout hosts' links are up
+    link_status_on_all_LC(duthosts, localhost, fanouts_and_ports)
+    
+    # Reboot RP should reboot both RP&LC, should detect all links on all linecards go down
+    dut_datetime = reboot(duthost, localhost, wait_for_ssh=False)
+
+    # RP doesn't have any interfaces, check all LCs' interfaces
+    link_status_on_all_LC(duthosts, localhost, fanouts_and_ports, up=False)
+
+    time.sleep(MAX_TIME_TO_REBOOT)
+
+    logger.info('reboot finished on {}, reboot started on {}'.format(hostname, dut_datetime))
+    dut_uptime = duthost.get_up_time()
+    logger.info('DUT {} up since {}'.format(hostname, dut_uptime))
+    assert float(dut_uptime.strftime("%s")) > float(dut_datetime.strftime("%s")), "Device {} did not reboot".format(hostname)
+
+    # After test, make sure all interfaces and services are up.
+    # This check is needed if chassis checked for RP then directly go to check for LCs, 
+    # w/o this check links may still be down even with time.sleep(MAX_TIME_TO_REBOOT)
+    check_interfaces_and_services(duthost, conn_graph_facts["device_conn"][hostname], xcvr_skip_list)
```

Comment:
should check_interfaces_and_services_all_LCs() be used here?

Reply:
thanks, done